pkg/osbuild/dnf_module_config_stage.go

```diff
@@ -0,0 +1,62 @@
+package osbuild
+
+import (
+	"github.com/osbuild/images/pkg/rpmmd"
+)
+
+type DNFModuleConfig struct {
+	Name     string   `json:"name,omitempty"`
+	Stream   string   `json:"stream,omitempty"`
+	State    string   `json:"state,omitempty"`
+	Profiles []string `json:"profiles,omitempty"`
+}
+
+type DNFModuleConfigStageOptions struct {
+	Config *DNFModuleConfig `json:"conf,omitempty"`
+}
+
+func (DNFModuleConfigStageOptions) isStageOptions() {}
+
+// NewDNFModuleConfigStageOptions creates a new DNFConfig Stage options object.
+func NewDNFModuleConfigStageOptions(config *DNFModuleConfig) *DNFModuleConfigStageOptions {
+	return &DNFModuleConfigStageOptions{
+		Config: config,
+	}
+}
+
+func (o DNFModuleConfigStageOptions) validate() error {
+	return nil
+}
+
+// NewDNFModuleConfigStage creates a new DNFModuleConfig Stage object.
+func NewDNFModuleConfigStage(options *DNFModuleConfigStageOptions) *Stage {
+	if err := options.validate(); err != nil {
+		panic(err)
+	}
+
+	return &Stage{
+		Type:    "org.osbuild.dnf.module-config",
+		Options: options,
+	}
+}
+
+func GenDNFModuleConfigStages(modules []rpmmd.ModuleSpec) []*Stage {
+	stages := make([]*Stage, 0)
```

Comment:
Nitpick: you could also `make([]*Stage, 0, len(modules))`, since the size is known in advance.

Reply:
Changed and force pushed.